autoload/neomake/makers/ft/rust.vim

```diff
@@ -64,6 +64,12 @@ function! neomake#makers#ft#rust#CargoProcessOutput(context) abort
         endif
 
         let span = data.spans[0]
+        for candidate_span in data.spans
+            if candidate_span.is_primary
+                let span = candidate_span
```

Comment:
`break`?

Reply:
According to https://github.com/rust-lang/rust.vim/issues/174#issuecomment-419472174 the first primary span should be used.